Source/Core/Core/PowerPC/Jit64/Jit_LoadStore.cpp

```diff
@@ -304,6 +303,15 @@ void Jit64::dcbz(UGeckoInstruction inst)
     AND(32, R(RSCRATCH), Imm32(~31));
   }
 
+  FixupBranch end_dcbz_hack;
+  if (SConfig::GetInstance().bLowDCBZHack)
+  {
+    // HACK: Don't clear any memory in the 0x80000000 - 0x80008000 region
+    LEA(32, RSCRATCH2, MDisp(RSCRATCH, 0x80000000));
+    CMP(32, R(RSCRATCH2), Imm32(0x7fff));
+    end_dcbz_hack = J_CC(CC_LE);
```

Comment:
CC_LE is a signed condition, I think you meant CC_BE. But since `0x8000'0000` is the smallest signed 32-bit integer, you can actually make use of that and get rid of the LEA (untested):

```suggestion
    // HACK: Don't clear any memory in the [0x8000'0000, 0x8000'8000) region.
    CMP(32, R(RSCRATCH), Imm32(0x8000'8000));
    end_dcbz_hack = J_CC(CC_L);
```
Neither GCC nor Clang seem to do this optimization so you might want to doublecheck...

(I also took the liberty of clarifying the interval boundaries.)

Reply:
Makes sense, and I tested that it doesn't break Cars 2.